index.js

```diff
@@ -2,11 +2,35 @@
 
 const fp = require('fastify-plugin')
 const helmet = require('helmet')
+const crypto = require('crypto')
 
 module.exports = fp(function (app, options, next) {
-  const middleware = helmet(options)
+  const generateNonces = !!options.generateNonces
+  delete options.generateNonces
+
+  const baseCspConfig = { contentSecurityPolicy: { directives: { defaultSrc: ["'self'"], scriptSrc: ["'self'"], styleSrc: ["'self'"] } } }
+
+  options = Object.assign({}, baseCspConfig, options)
+
+  let middleware = helmet(options)
 
   app.addHook('onRequest', function (req, reply, next) {
+    if (generateNonces) {
+      const nonce = crypto.randomBytes(16).toString('base64')
+
+      options.contentSecurityPolicy.directives.scriptSrc.push(`'${nonce}'`)
+      options.contentSecurityPolicy.directives.styleSrc.push(`'${nonce}'`)
+
+      if (!reply.raw.locals) {
+        reply.raw.locals = {}
+      }
+
+      // TODO - what is the best place to put this?
+      reply.raw.locals.nonce = nonce
```

Comment:
This is actually a deeper question for me - I still haven't learned what the "fastify" equivalent way of doing what I used to do in express with `res.locals`
What is the best place to put these nonces? (and a general question from me - where would you generally store "middleware" mutation type data?)

Reply:
What is `res.locals`?